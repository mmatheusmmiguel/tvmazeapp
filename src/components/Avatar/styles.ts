import styled from 'styled-components/native';

export const AvatarContainer = styled.View`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

export const AvatarNameText = styled.Text`
    display: flex;
    color: ${props => (props.fontColor ? props.fontColor : '#fff')}
    font-size: ${props => (props.fontSize ? props.fontSize + 'px' : '15px')}
    font-weight: 500;
`;
export const AvatarDescriptionText = styled.Text`
    display: flex;
    color: ${props => (props.fontColor ? props.fontColor : '#ccc')}
    font-size: ${props => (props.fontSize ? props.fontSize + 'px' : '14px')}
    font-weight: 400;
`;
export const AvatarIcon = styled.Image`
    height: ${props => (props.imageSize ? props.imageSize + 'px' : '90px')}
    width: ${props => (props.imageSize ? props.imageSize + 'px' : '90px')}
    margin: 10px;
    border-radius: ${props =>
      props.imageSize ? props.imageSize / 2 + 'px' : '45px'}
`;
