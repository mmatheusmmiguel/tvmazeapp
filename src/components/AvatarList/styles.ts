import styled from 'styled-components/native';

export const AvatarScrollView = styled.ScrollView`
  margin-bottom: 20px;
  flex-direction: ${props => (props.isHorizontal ? 'row' : 'column')};
`;
