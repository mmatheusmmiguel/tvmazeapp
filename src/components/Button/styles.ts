import styled from 'styled-components/native';
import constants from '../../constants';

export const ButtonContainer = styled.TouchableOpacity`
    padding: 14px;
    justify-content: center;
    flex-direction: row;
    background-color: ${props => props.backgroundColor ? props.backgroundColor : constants.COLORS.RED}
    margin: 10px;
    border-radius: 5px;
    align-items: center;
`

export const ButtonTextBold = styled.Text`
    margin: 2px;
    font-size: ${props => props.fontSize ? props.fontSize + "px" : "18px"};
    font-weight: bold;
    color: ${props => props.fontColor ? props.fontColor : constants.COLORS.WHITE};
`