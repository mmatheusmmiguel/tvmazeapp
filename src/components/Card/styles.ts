import styled from 'styled-components/native';

export const CardView = styled.View`
  padding: 15px;
  background-color: #1e1e21;
  margin: 5px;
  border-radius: 10px;
`;

//fontSize: 20, fontWeight: 'bold', color: '#FFF'
export const CardTitle = styled.Text`
    font-size: 20px;
    font-weight: bold;
    color: #FFF;
`