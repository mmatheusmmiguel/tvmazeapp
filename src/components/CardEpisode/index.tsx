import React from 'react';
import {EpisodeType} from '../../@types';
import {
  CardEpisodeContainer,
  CardImageBackground,
  DurationText,
  DurationTextContainer,
  EpisodeNumberText,
  ImageOverlayView,
  TitleText,
} from './styles';

interface CardEpisodeProps {
  item: EpisodeType;
  imageOverlayOpacity?: number;
  onPress: Function;
}

export function CardEpisode({
  item,
  onPress,
  imageOverlayOpacity,
}: CardEpisodeProps) {
  return (
    <CardEpisodeContainer onPress={onPress}>
      <CardImageBackground
        source={{uri: item.image?.original}}
        imageStyle={{borderRadius: 8}}>
        <ImageOverlayView opacity={imageOverlayOpacity} />
        <DurationTextContainer>
          <DurationText>{item.runtime} min</DurationText>
        </DurationTextContainer>
      </CardImageBackground>
      <EpisodeNumberText>EPISODE {item.number}</EpisodeNumberText>
      <TitleText>{item.name}</TitleText>
    </CardEpisodeContainer>
  );
}

export default CardEpisode;
