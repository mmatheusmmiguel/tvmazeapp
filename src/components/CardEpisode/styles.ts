import { TouchableOpacity } from 'react-native';
import styled from 'styled-components/native';

export const CardEpisodeContainer: TouchableOpacity = styled.TouchableOpacity`
  margin-right: 20px;
  flex-direction: column;
  align-items: flex-start;
`;
export const CardImageBackground = styled.ImageBackground`
  height: 110px;
  width: 200px;
`;
export const ImageOverlayView = styled.View`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: #000;
  opacity: ${props => (props.opacity ? props.opacity : '0.3')};
`;
export const DurationTextContainer = styled.View`
  position: absolute;
  left: 0;
  bottom: 0;
`;
export const DurationText = styled.Text`
  text-shadow: 2px 2px 5px #000;
  padding: 5px;
  font-size: 16px;
  font-weight: 500;
  color: #fff;
`;
export const EpisodeNumberText = styled.Text`
  font-size: 16px;
  margin-top: 5px;
  color: #ccc;
`;
export const TitleText = styled.Text`
  font-size: 18px;
  font-weight: bold;
  color: #fff;
  width: 200px;
`;
