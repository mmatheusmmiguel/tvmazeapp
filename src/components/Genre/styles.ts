import styled from 'styled-components/native';
import constants from '../../constants';

export const GenreContainer = styled.View`
  border-radius: 20px;
  margin: 5px;
  padding: 10px;
  background-color: ${props =>
    props.backgroundColor ? props.backgroundColor : constants.COLORS.RED};
`;

export const GenreText = styled.Text`
font-size: ${props => (props.fontSize ? props.fontSize : '16px')}
font-weight: bold;
color: ${props => (props.fontColor ? props.fontColor : '#fff')}
`;
