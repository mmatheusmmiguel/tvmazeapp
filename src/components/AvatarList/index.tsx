import React from 'react';
import {IGetCast} from '../../services/Cast';
import Avatar from '../Avatar';
import {AvatarScrollView} from './styles';

interface AvatarListProps {
  horizontal?: boolean;
  dataSource: IGetCast[];
}

export function AvatarList({horizontal, dataSource}: AvatarListProps) {
  return (
    <AvatarScrollView horizontal={horizontal} isHorizontal={horizontal}>
      {dataSource?.length &&
        dataSource.map((item: IGetCast, index) => (
          <Avatar
            key={index}
            url={item.person.image?.original}
            name={item.person?.name}
            description={item.character?.name}
          />
        ))}
    </AvatarScrollView>
  );
}

export default AvatarList;
