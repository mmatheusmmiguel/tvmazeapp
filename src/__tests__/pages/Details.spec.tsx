import React from 'react';
import {
  fireEvent,
  render,
  RenderResult,
  screen,
  waitFor,
} from '@testing-library/react-native';
import Details from '../../pages/Details';
import 'jest-styled-components';
import {HeaderImage, Title} from '../../pages/Details/styles';
import constants from '../../constants';
import AvatarList from '../../components/AvatarList';
import {IGetCast} from '../../services/Cast';
import {EpisodeType} from '../../types';
import CardEpisode from '../../components/CardEpisode';

const mockedDispatch = jest.fn();

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: mockedDispatch,
    }),
  };
});

jest.mock(
  'react-native/Libraries/Components/Touchable/TouchableOpacity.js',
  () => {
    const { View } = require('react-native')
    const MockTouchable = props => {
      return <View {...props} />
    }
    MockTouchable.displayName = 'TouchableOpacity'

    return MockTouchable
  }
)

describe('Details Component', () => {
  beforeEach(() => {
    mockedDispatch.mockClear();
  });

  it('Should render Header Component', async () => {
    const TVShowName: string = 'The Powerpuff Girls';

    const headerImage = render(
      <HeaderImage
        source={{
          uri: 'https://static.tvmaze.com/uploads/images/original_untouched/60/151357.jpg',
        }}
        height={constants.IMAGE_HEADER_HEIGHT}
      />,
    ).toJSON();

    const headerTitle = render(<Title>{TVShowName}</Title>);

    expect(headerImage).toBeTruthy();
    expect(headerTitle.getByText(TVShowName)).toBeTruthy();
  });

  it('Should has Cast List and render', () => {
    const castList: IGetCast[] = [
      {
        person: {
          id: 1,
          url: '',
          name: 'Matheus',
          birthday: '1999-05-29',
          gender: 'Male',
          country: {
            name: 'Brazil',
            code: 'BR',
          },
          image: {
            medium: '',
            original: '',
          },
        },
        character: {
          id: 1,
          url: '',
          name: 'Matheus',
          image: '',
        },
      },
    ];
    const {queryAllByText} = render(<AvatarList dataSource={castList} />);

    expect(castList.length).toBeGreaterThan(0);
    expect(screen.queryAllByText('Matheus')).toBeTruthy();
  });

  it('Should has a Episode List', async () => {
    const navigate = jest.fn();
    
    const episodesList: EpisodeType[] = [
      {
        id: 1,
        image: {
          original: '',
          medium: '',
        },
        name: '',
        number: 1,
        rating: {
          average: 10,
        },
        runtime: 10,
        season: 1,
        summary: 'Teste',
      },
    ];
    expect(episodesList.length).toBeGreaterThan(0);
    const cardEpisode: RenderResult = render(
      <CardEpisode
        key={episodesList[0].id}
        title={episodesList[0].name}
        cardImage={episodesList[0].image?.original}
        number={episodesList[0]?.number}
        duration={episodesList[0]?.runtime}
        onPress={() => {
          expect(screen.toJSON()).toMatchSnapshot()
        }}
      />
    );

    await waitFor(() => expect(cardEpisode).toBeTruthy())
    fireEvent.press(cardEpisode.toJSON());
  });
});
