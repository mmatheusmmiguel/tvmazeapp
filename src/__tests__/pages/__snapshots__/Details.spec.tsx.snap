// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Component Should has a Episode List 1`] = `
<View
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "flexDirection": "column",
        "marginRight": 20,
      },
    ]
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Array [
        Object {
          "height": 110,
          "width": 200,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 110,
            "width": 200,
          },
          Object {
            "borderRadius": 8,
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#000",
            "bottom": 0,
            "left": 0,
            "opacity": 0.3,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 16,
              "fontWeight": "500",
              "paddingBottom": 5,
              "paddingLeft": 5,
              "paddingRight": 5,
              "paddingTop": 5,
              "textShadowColor": "#000",
              "textShadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "textShadowRadius": 5,
            },
          ]
        }
      >
        10
         min
      </Text>
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#ccc",
          "fontSize": 16,
          "marginTop": 5,
        },
      ]
    }
  >
    EPISODE 
    1
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "bold",
          "width": 200,
        },
      ]
    }
  >
    
  </Text>
</View>
`;
